### ISSUE TITLE
Zulip python api requires matrix_client as a dependency

### ISSUE THREAD

**Issue by aleksanb** [zulip/python-zulip-api#751]
Today you get matrix_client in your deps if you install zulip.
Perhaps this should be an optional feature instead, so that if you install zulip[matrix_client] you'll get it, but otherwise not?
My use case is shipping error messages from a server to zulip using zulip.send_message only, so the matrix dep is unneeded.
This should be the case for other users as well.

**Comment by rht** [comment]
Yeah, I think we want `matrix_client` as an optional dependency, installed via `pip install zulip[matrix]`.
